src/net.cpp

```diff
@@ -1877,8 +1877,8 @@ void CConnman::ThreadOpenConnections(const std::vector<std::string> connect)
 
             // Checking !dnsseed is cheaper before locking 2 mutexes.
             if (!add_fixed_seeds_now && !dnsseed) {
-                LOCK2(m_addr_fetches_mutex, cs_vAddedNodes);
-                if (m_addr_fetches.empty() && vAddedNodes.empty()) {
+                LOCK2(m_addr_fetches_mutex, m_added_nodes_mutex);
+                if (m_addr_fetches.empty() && m_added_nodes.empty()) {
                     add_fixed_seeds_now = true;
                     LogPrintf("Adding fixed seeds as -dnsseed=0, -addnode is not provided and all -seednode(s) attempted\n");
                 }
```

Comment:
Not necessary for this PR, but it is a good practice to avoid printouts (which may block) while holding mutexes. Feel free to take this or ignore it:
```diff
-                LOCK2(m_addr_fetches_mutex, m_added_nodes_mutex);
-                if (m_addr_fetches.empty() && m_added_nodes.empty()) {
-                    add_fixed_seeds_now = true;
+                {
+                    LOCK2(m_addr_fetches_mutex, m_added_nodes_mutex);
+                    add_fixed_seeds_now = m_addr_fetches.empty() && m_added_nodes.empty();
+                }
+                if (add_fixed_seeds_now) {
                     LogPrintf("Adding fixed seeds as -dnsseed=0, -addnode is not provided and all -see
                 }
```